src/relay/backend/contrib/tensorrt/enable_tensorrt.cc

```diff
@@ -0,0 +1,569 @@
+        {"power", AlwaysChecker},
+        {"nn.max_pool2d", MaxPool2DOpChecker},
+        {"nn.avg_pool2d", AvgPool2DOpChecker},
+        {"nn.global_max_pool2d", GlobalPool2DOpChecker},
+        {"nn.global_avg_pool2d", GlobalPool2DOpChecker},
+        {"exp", AlwaysChecker},
+        {"log", AlwaysChecker},
+        {"sqrt", AlwaysChecker},
+        {"abs", AlwaysChecker},
+        {"negative", AlwaysChecker},
+        {"nn.batch_flatten", AlwaysChecker},
+        {"expand_dims", ExpandDimsOpChecker},
+        {"squeeze", SqueezeOpChecker},
+        {"concatenate", ConcatenateOpChecker},
+        {"nn.conv2d_transpose", Conv2DTransposeOpChecker},
+        {"transpose", TransposeOpChecker},
+        {"reshape", ReshapeOpChecker},
+        {"nn.pad", PadOpChecker},
+        {"sum", ReduceOpChecker},
+        {"prod", ReduceOpChecker},
+        {"max", ReduceOpChecker},
+        {"min", ReduceOpChecker},
+        {"mean", ReduceOpChecker},
+        {"contrib.adaptive_max_pool2d", AdapativePool2DOpChecker},
+        {"contrib.adaptive_avg_pool2d", AdapativePool2DOpChecker},
+        // Ops which require TRT 5.1.5+
+        {"clip", TrtVersionChecker<5, 1, 5>},
+        {"nn.leaky_relu", TrtVersionChecker<5, 1, 5>},
+        {"sin", TrtVersionChecker<5, 1, 5>},
+        {"cos", TrtVersionChecker<5, 1, 5>},
+        {"atan", TrtVersionChecker<5, 1, 5>},
+        {"ceil", TrtVersionChecker<5, 1, 5>},
+        {"floor", TrtVersionChecker<5, 1, 5>},
+        {"strided_slice", StridedSliceOpChecker},
+        // Ops which require TRT 6.0.1+
+        {"image.resize", ResizeOpChecker}};
+
+// TODO(trevmorr): Use annotation/partitioning framework when that is available.
+/*!
+ * \brief Visitor to check whethere a relay Expr is compatible with a certain
+ * version of TensorRT.
+ */
+class TrtChecker : public ExprVisitor {
+ public:
+  explicit TrtChecker(const std::tuple<int, int, int>& trt_version)
+      : trt_version_(trt_version) {}
+
+  void VisitExpr_(const VarNode* op) {
+    const auto* ttype = op->checked_type().as<TensorTypeNode>();
+    CHECK(ttype);
+    if (!ttype->dtype.is_float()) {
+      compatible_ = false;
+    }
+  }
+
+  void VisitExpr_(const CallNode* call) final {
+    const std::string op_name = (call->op.as<OpNode>())->name;
+    for (size_t i = 0; i < call->args.size(); ++i) {
+      // Workaround for check for transpose of weight. This check allows weights
+      // to be either 1) VarNode 2) ConstantNode 3) VarNode or Constant Node
+      // with a transpose between. This transpose is allowed to have permuation
+      // which isn't allowed by TensorRT since we will process this transpose
+      // manually.
+      // This workaround isn't required anymore since ConstantFolding will take
+      // care of the transpose for us. However, in the case where the weights
+      // aren't marked as params it can still be useful.
+      if ((op_name == "nn.conv2d" || op_name == "nn.dense") && i == 1) {
```

Comment:
I'd put `i == 1` expr to the left of `&&` because it can prevent unnecessary string comparisons